File: MagniUniversity.UI/Controllers/GradeController.cs
using MagniUniversity.Domain.Model;
using MagniUniversity.Service.Interface;
using Newtonsoft.Json;
using System;
using System.Web.Mvc;

namespace MagniUniversity.UI.Controllers
{
    public class GradeController : Controller
    {
        private readonly IEnrollmentService _service;

        public GradeController(IEnrollmentService service)
        {
            _service = service;
        }

        [HttpGet]
        public JsonResult Index()
        {
            try
            {
                var list = _service.List();
                return Json(list, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                Response.StatusCode = 500;
                return Json(new { error_message = "Error: " + ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        [HttpPost]
        public JsonResult Save(Enrollment model)
        {
            try
            {
                model.Student = null;
                model.Subject = null;
                var retModel = _service.Save(model);
                return Json(new { Status = "Ok" }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                Response.StatusCode = 500;
                return Json(new { error_message = "Error: " + ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        [HttpDelete]
        public JsonResult Remove(int id)
        {
            try
            {
                _service.Remove(id);
                return Json(new { Status = "Ok" }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                Response.StatusCode = 500;
                return Json(new { error_message = "Error: " + ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
    }
}

File: MagniUniversity.UI/app/controllers/gradeController.js
'use strict';

app.controller('gradeController', ['$scope', 'crudBaseService', function ($scope, crudBaseService) {
    
    $scope.objectList = [];
    $scope.msgError = "";

    crudBaseService.SetAlias('grade');

    $scope.GetList = function () {
        
        crudBaseService.GetList().then(function (results) {
            if (results.data.error_message) {
                $scope.msgError = results.data.error_message;
            } else {
                $scope.objectList = results.data;
            }
        }, function (error) {
            $scope.msgError = error;
        });
    };

    $scope.Save = function (enrollment) {

        $scope.msgError = '';
        var objForm = enrollment;

        if ($scope.msgError.length === 0) {
            crudBaseService.Save(objForm).then(function (results) {
                if (results.data.error_message) {
                    $scope.msgError = results.data.error_message;
                } else {
                    bootbox.alert("Success!");
                    $scope.GetList();
                    _setDefaultModel();
                }
            }, function (error) {
                $scope.msgError = error;
            });
        }

    };

    $scope.Remove = function (id) {

        crudBaseService.Remove(id).then(function (results) {
            if (results.data.error_message) {
                $scope.msgError = results.data.error_message;
            } else {
                bootbox.alert("Success!");
                $scope.GetList();
            }
        }, function (error) {
            $scope.msgError = error;
        });

    };

    $scope.GetList();

}]);